src/backend/distributed/commands/trigger.c

```diff
@@ -203,45 +214,602 @@ get_relation_trigger_oid_compat(HeapTuple heapTuple)
+ * PostprocessCreateTriggerStmt is called after a CREATE TRIGGER command has
+ * been executed by standard process utility. This function errors out for
+ * unsupported commands or creates ddl job for supported CREATE TRIGGER commands.
+ */
+List *
+PostprocessCreateTriggerStmt(Node *node, const char *queryString)
+{
+	CreateTrigStmt *createTriggerStmt = castNode(CreateTrigStmt, node);
+
+	RangeVar *relation = createTriggerStmt->relation;
+
+	bool missingOk = false;
+	Oid relationId = RangeVarGetRelid(relation, CREATE_TRIGGER_LOCK_MODE, missingOk);
+
+	if (!IsCitusTable(relationId))
+	{
+		return NIL;
+	}
+
+	ErrorIfUnsupportedCreateTriggerCommand(createTriggerStmt);
+
+	if (IsCreateCitusTruncateTriggerStmt(createTriggerStmt))
+	{
+		return NIL;
+	}
+
+	if (IsCitusLocalTable(relationId))
+	{
+		ObjectAddress objectAddress = GetObjectAddressFromParseTree(node, missingOk);
+		EnsureDependenciesExistOnAllNodes(&objectAddress);
+
+		char *triggerName = createTriggerStmt->trigname;
+		return CitusLocalTableTriggerCommandDDLJob(relationId, triggerName,
+												   queryString);
+	}
+
+	return NIL;
+}
+
+
+/*
+ * CreateTriggerStmtObjectAddress finds the ObjectAddress for the trigger that
+ * is created by given CreateTriggerStmt. If missingOk is false and if trigger
+ * does not exist, then it errors out.
  *
- * Note that internal triggers that are created implicitly by postgres for
- * foreign key validation already wouldn't be executed via process utility,
- * hence there is no need to check that case here.
+ * Never returns NULL, but the objid in the address can be invalid if missingOk
+ * was set to true.
  */
-void
+ObjectAddress
+CreateTriggerStmtObjectAddress(Node *node, bool missingOk)
+{
+	CreateTrigStmt *createTriggerStmt = castNode(CreateTrigStmt, node);
+
+	RangeVar *relation = createTriggerStmt->relation;
+	Oid relationId = RangeVarGetRelid(relation, CREATE_TRIGGER_LOCK_MODE, missingOk);
+
+	char *triggerName = createTriggerStmt->trigname;
+	Oid triggerId = get_trigger_oid(relationId, triggerName, missingOk);
+
+	if (triggerId == InvalidOid && missingOk == false)
+	{
+		char *relationName = get_rel_name(relationId);
+		ereport(ERROR, (errcode(ERRCODE_UNDEFINED_OBJECT),
+						errmsg("trigger \"%s\" on relation \"%s\" does not exist",
+							   triggerName, relationName)));
+	}
+
+	ObjectAddress address = { 0 };
+	ObjectAddressSet(address, TriggerRelationId, triggerId);
+	return address;
+}
+
+
+/*
+ * IsCreateCitusTruncateTriggerStmt returns true if given createTriggerStmt
+ * creates citus_truncate_trigger.
+ */
+static bool
+IsCreateCitusTruncateTriggerStmt(CreateTrigStmt *createTriggerStmt)
+{
+	List *functionNameList = createTriggerStmt->funcname;
+	RangeVar *functionRangeVar = makeRangeVarFromNameList(functionNameList);
+	char *functionName = functionRangeVar->relname;
+	if (strncmp(functionName, CITUS_TRUNCATE_TRIGGER_NAME, NAMEDATALEN) == 0)
+	{
+		return true;
```

Comment:
It seems that we never return true from this method in our test coverage, can you add a test to hit this line?

Reply:
Actually, we hit to this line in upgrade tests, I know it because if I remove this check, upgrade tests fail.
I suspect that codecov measures the patch coverage "not considering" the existing tests